builtin/logical/ssh/path_issuers.go

```diff
@@ -0,0 +1,517 @@
+			},
+		},
+		HelpSynopsis:    pathIssuersSyn,
+		HelpDescription: pathIssuersDesc,
+	}
+}
+
+func pathSubmitIssuer(b *backend) *framework.Path {
+	fields := map[string]*framework.FieldSchema{}
+	fields = addSubmitIssuerCommonFields(fields)
+
+	fields["set_default"] = &framework.FieldSchema{
+		Type:        framework.TypeBool,
+		Description: `If true, this issuer will be set as the default issuer for performing operations. Only one issuer can be the default issuer and, if there's one set, it will be overridden.`,
+		Default:     false,
+	}
+
+	return &framework.Path{
+		Pattern: "issuers/import" + framework.OptionalGenericNameRegex("issuer_name"),
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixSSH,
+			OperationVerb:   "submit",
+			OperationSuffix: "issuer",
+		},
+
+		Fields: fields,
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathWriteIssuerHandler,
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationVerb:   "submit",
+					OperationSuffix: "issuer",
+				},
+				Responses: map[int][]framework.Response{
+					http.StatusOK: {
+						{
+							Description: "OK",
+							Fields:      issuerOKResponseFields,
+						},
+					},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathSubmitIssuerSyn,
+		HelpDescription: pathSubmitIssuerDesc,
+	}
+}
+
+func pathListIssuers(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "issuers/?$",
+
+		DisplayAttrs: &framework.DisplayAttributes{
+			OperationPrefix: operationPrefixSSH,
+			OperationVerb:   "list",
+			OperationSuffix: "issuers",
+		},
+
+		Fields: map[string]*framework.FieldSchema{
+			"after": {
+				Type:        framework.TypeString,
+				Required:    false,
+				Description: `Optional entry to list begin listing after, not required to exist.`,
+			},
+			"limit": {
+				Type:        framework.TypeInt,
+				Required:    false,
+				Description: `Optional number of entries to return; defaults to all entries.`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ListOperation: &framework.PathOperation{
+				Callback: b.pathListIssuersHandler,
+				Responses: map[int][]framework.Response{
+					http.StatusOK: {{
+						Description: "OK",
+						Fields: map[string]*framework.FieldSchema{
+							"keys": {
+								Type:        framework.TypeStringSlice,
+								Description: `A list of keys`,
+								Required:    true,
+							},
+							"key_info": {
+								Type:        framework.TypeMap,
+								Description: `Key info with issuer identifier`,
+								Required:    false,
+							},
+						},
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathListIssuersSyn,
+		HelpDescription: pathListIssuersDesc,
+	}
+}
+
+func pathGetIssuerPublicKeyUnauthenticated(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "issuer/" + framework.GenericNameRegex("issuer_ref") + "/public_key",
```

Comment:
Is this strictly a raw endpoint or do we have JSON endpoints like PKI does?

Reply:
Didn't see what PKI did here, I have based myself more on what we had before. This is raw, unauthenticated, returning only the public key in plain text format. 

Then there's the authenticated endpoint which returns a JSON response with the fields `issuer_id`, `issuer_name` and `public_key`.

Of course, both of these accept an issuer reference or `default`. I'm going to see how does this align with PKI.

Reply:
Good enough for now. We can revisit and add a JSON endpoint in the future. Thanks!